ngx-fudis/projects/ngx-fudis/src/lib/components/typography/heading/heading.component.ts

```diff
@@ -46,19 +51,46 @@ export class HeadingComponent implements OnInit {
 	protected _id: string;
 
 	getHeadingMarginBottom(): string {
-		if (this.size === 'xxl' || this.size === 'xl') {
+		if (this.level === 1 || this.level === 2) {
 			return 'sm';
 		}
 		return 'xs';
 	}
 
+	getHeadingSize(): string {
+		switch (this.level) {
+			case 1:
+				return 'xxl';
+			case 2:
+				return 'xl';
+			case 3:
+				return 'lg';
+			case 4:
+				return 'md';
+			case 5:
+				return 'sm';
+			case 6:
+				return 'xs';
+			default:
+				return 'lg';
+		}
+	}
+
 	ngOnInit(): void {
 		this._id = this.id ?? this._idService.getNewId('heading');
 
+		if (this.size) {
+			this._classList = `fudis-heading fudis-heading__${this.size}`;
+		} else {
+			this._classList = `fudis-heading fudis-heading__${this.getHeadingSize()}`;
+		}
+
 		if (this.marginBottom) {
-			this._classList = `fudis-heading fudis-heading__${this.size} fudis-mb-${this.marginBottom}`;
+			this._classList += ` fudis-mb-${this.marginBottom}`;
 		} else {
-			this._classList = `fudis-heading fudis-heading__${this.size} fudis-mb-${this.getHeadingMarginBottom()}`;
+			this._classList += ` fudis-mb-${this.getHeadingMarginBottom()}`;
 		}
+
+		this._classList += ` fudis-heading__${this.align}`;
```

Comment:
This classlist generation should be own function e. g. _setClasses

Also generated classlist would be cleaner, if instead of strings with spaces it would be just a array of strings. It would make this cleaner.